File: Find_Players_With_Zero_or_One_Losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        vector<vector<int>> res(2, vector<int>());
        unordered_map<int, int> loseTimes;

        for (const auto& match : matches) {
            int w = match[0];
            int l = match[1];

            loseTimes[w] = loseTimes[w];
            loseTimes[l] = 1 + loseTimes[l];
        }

        for (const auto& entry : loseTimes) {
            int player = entry.first;
            int loseCount = entry.second;

            if (loseCount < 2) {
                res[loseCount].push_back(player);
            }
        }

        sort(res[0].begin(), res[0].end());
        sort(res[1].begin(), res[1].end());

        return res;        
    }
};

File: Find_Players_With_Zero_or_One_Losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        List<List<Integer>> res = new ArrayList<>();
        res.add(new ArrayList<>());
        res.add(new ArrayList<>());
        Map<Integer, Integer> loseTimes = new HashMap<>();

        for (int[] match : matches) {
            int w = match[0];
            int l = match[1];

            loseTimes.put(w, loseTimes.getOrDefault(w, 0));
            loseTimes.put(l, 1 + loseTimes.getOrDefault(l, 0));
        }

        for (Map.Entry<Integer, Integer> entry : loseTimes.entrySet()) {
            int player = entry.getKey();
            int loseCount = entry.getValue();

            if (loseCount < 2) {
                res.get(loseCount).add(player);
            }
        }

        Collections.sort(res.get(0));
        Collections.sort(res.get(1));

        return res;        
    }
}
